src/MSBuild/XMake.cs

```diff
@@ -211,14 +211,14 @@ private static void HandleConfigurationException(Exception ex)
         /// MSBuild no longer runs any arbitrary code (tasks or loggers) on the main thread, so it never needs the
         /// main thread to be in an STA. Accordingly, to avoid ambiguity, we explicitly use the [MTAThread] attribute.
         /// This doesn't actually do any work unless COM interop occurs for some reason.
-        /// We use the MultiDomainHost loader policy because we may create secondary AppDomains and need NGEN images
-        /// for Framework / GACed assemblies to be loaded domain neutral so their native images can be used.
+        /// We use the MultiDomain loader policy because we may create secondary AppDomains and need NGEN images
+        /// for our as well as Framework assemblies to be loaded domain neutral so their native images can be used.
         /// See <see cref="NuGetFrameworkWrapper"/>.
         /// </remarks>
         /// <returns>0 on success, 1 on failure</returns>
         [MTAThread]
 #if FEATURE_APPDOMAIN
-        [LoaderOptimization(LoaderOptimization.MultiDomainHost)]
+        [LoaderOptimization(LoaderOptimization.MultiDomain)]
```

Comment:
This may have a negative perf impact on steady state performance because statics are accessed with an extra indirection in domain-neutral code. I have run an evaluation micro-benchmark and it's not shown any regression but still, this is somewhat unfortunate.

Ideally we would revert the secondary AD logic altogether and instead load the assembly into the default AD by strong name. @jeffkl, you've mentioned that the NuGet SDK resolver may have its version frozen to allow this. Do you think it can realistically happen to `NuGet.Frameworks.dll` - which we use as a library to implement a few property functions?

Reply:
This sounds not great but I would go ahead with it. I'm not sure I understand the docs on this though--it sounds like we wouldn't have gotten much from `MultiDomainHost` because none of our assemblies or our main references were GACed anyway?

Reply:
Correct. When testing the original change I was fixated on making sure that `NuGet.Frameworks` can use its native image and for _its_ dependencies `MultiDomainHost` was enough. I missed the obvious fact that it was not enough for `Microsoft.Build`.